elastic/kibana
# [dashboard] wrap presentation panel in error boundary so buggy panel does not prevent rest of dashboard from rendering

React embeddables no longer render in their own react tree. This means that a buggy react embeddable will prevent the entire dashboard from rendering. This PR resolves this problem by putting an error boundary between react embeddables and the dashboard.

### test instructions
1. start kibana with `yarn start --run-examples`
2. Update eui_markdown_react_embeddable, putting `throw new Error('simulated failure');` before returning `EuiMarkdown`
3. Create a dashboard with EuiMarkdownEmbeddable, verify all dashboard panels render

#### Before EuiErrorBoundary
<img width="500" alt="Screenshot 2024-02-15 at 2 23 58 PM" src="https://github.com/elastic/kibana/assets/373691/90eeb522-df99-4da0-84fa-7876b6b7f304">

#### After EuiErrorBoundary
<img width="500" alt="Screenshot 2024-02-15 at 2 23 30 PM" src="https://github.com/elastic/kibana/assets/373691/8c7c42fa-c246-4da5-bf90-0d1abb19cc09">
